packages/cli/source/manifest/compiler.ts

```diff
@@ -40,188 +67,45 @@ export const logDuplicateError = (
-    const result: Query = {
-        name: "",
-        description: "",
-        resource: "",
-        content: "",
-    };
-
-    const missingKeys = getMissingKeys(constants.queryKeys, query);
-
-    if (missingKeys) {
-        logMissingKeys("queries", missingKeys, path);
-    }
-
-    for (const key in query) {
-        const value = (query as any)[key];
-
-        switch (key) {
-            case "name": {
-                if (!IDENTIFIER_REGEX.test(value)) {
-                    logSemanticError("Query name is invalid.", path);
-                }
-                result.name = value.trim();
-                break;
-            }
-
-            case "description": {
-                result.description = value.trim();
-                break;
-            }
-
-            case "resource": {
-                if (!IDENTIFIER_REGEX.test(value)) {
-                    logSemanticError("Query resource is invalid.", path);
-                }
-                result.resource = value.trim();
-                break;
-            }
-
-            case "content": {
-                result.content =
-                    typeof value === "string" ? value.trim() : value;
-                break;
-            }
-        }
-    }
-    return result;
-};
-
-const parseResource = (resource: Resource, path: string): Resource => {
-    const result: Resource = {
-        name: "",
-        description: "",
-        type: "",
-        connection: "",
-    };
-
-    const missingKeys = getMissingKeys(constants.resourceKeys, resource);
-
-    if (missingKeys) {
-        logMissingKeys("resources", missingKeys, path);
-    }
-
-    for (const key in resource) {
-        const value = (resource as any)[key];
-
-        switch (key) {
-            case "name": {
-                if (!IDENTIFIER_REGEX.test(value)) {
-                    logSemanticError("Resource name is invalid.", path);
-                }
-                result.name = value.trim();
-                break;
-            }
-
-            case "description": {
-                result.description = value.trim();
-                break;
-            }
-
-            case "type": {
-                result.type = value.trim();
-                break;
-            }
-
-            // TODO: Implement parseConnection()
-            case "connection": {
-                result.connection =
-                    typeof value === "string" ? value.trim() : value;
-                break;
-            }
-        }
-    }
-    return result;
-};
-
-const parseQueries = (queries: any, path = "<anonymous>") => {
+const validateQueries = (queries: any, path = "<anonymous>") => {
     const result: any = {};
     for (const query of queries) {
         if (result[query.name]) {
             logDuplicateError(query.name, path);
         }
-        result[query.name] = parseQuery(query, path);
+        const { error, value } = querySchema.validate(query, {
```

Comment:
By default, Joi returns one error at a time. It would be cumbersome to fix all the errors. You need to specify `abortEarly: false` to get a list of all the errors in one shot. Read more about this in the docs.